From 1c4d31b99afaa6121b4265783f70b6cc9951b786 Mon Sep 17 00:00:00 2001
From: Donald Carr <d@chaos-reins.com>
Date: Tue, 13 Jun 2017 11:19:30 -0700
Subject: [PATCH] Avoid double deletion of mBuffer

Both WaylandBrcmBuffer and WaylandBuffer were calling

wl_buffer_destroy(mBuffer)

without the subclass nulling the pointer on deletion

Change-Id: I66a553a3d11a1ef90bbfe6ad0fb49f40fe1a0bdb
---
 src/hardwareintegration/client/brcm-egl/qwaylandbrcmeglwindow.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/hardwareintegration/client/brcm-egl/qwaylandbrcmeglwindow.cpp b/src/hardwareintegration/client/brcm-egl/qwaylandbrcmeglwindow.cpp
index 9842f61c..f2709983 100644
--- a/src/hardwareintegration/client/brcm-egl/qwaylandbrcmeglwindow.cpp
+++ b/src/hardwareintegration/client/brcm-egl/qwaylandbrcmeglwindow.cpp
@@ -91,6 +91,7 @@ public:
     {
         wl_array_release(&m_array);
         wl_buffer_destroy(mBuffer);
+        mBuffer = nullptr;
     }
 
     QSize size() const { return m_size; }
-- 
2.13.1

